include/deal.II/base/bounding_box.h

```diff
@@ -432,6 +469,19 @@ inline BoundingBox<spacedim, Number>::BoundingBox(
 
 
 
+template <int spacedim, typename Number>
+inline bool
+BoundingBox<spacedim, Number>::is_valid() const
+{
+  for (unsigned int i = 0; i < spacedim; ++i)
+    if (this->boundary_points.first[i] > this->boundary_points.second[i])
```

Comment:
Isn't it an invariant of the class that that can't happen? I can see from your new constructor that it can happen that `.first == .second`, but I thought that this case can't happen.